crossbar/worker/proxy.py

```diff
@@ -60,6 +57,284 @@ class ProxyWorkerProcess(worker.NativeWorkerProcess):
     LOGNAME = 'Proxy'
 
 
+class ProxySession(object):
+    """
+    A router-side proxy session that handles incoming client
+    connections.
+    """
+
+    def __init__(self, router_factory):
+        self.transport = None
+        self._router_factory = router_factory
+
+        # basic session information
+        self._pending_session_id = None
+        self._realm = None
+        self._session_id = None
+        self._session_roles = None
+        self._session_details = None
+
+        # session authentication information
+        self._pending_auth = None
+        self._authid = None
+        self._authrole = None
+        self._authmethod = None
+        self._authprovider = None
+        self._authextra = None
+        self._controller = router_factory._proxy_controller
+
+        # if we have a backend connection, it'll be here (and be a
+        # Session instance)
+        self._backend_session = None
+
+        # in _transport_info are (possibly) the following keys (gleaned from router/session.py):
+        # client_cert
+        # channel_id
+        # type (websocket, etc)
+        # protocol (websocket subprotocol?) None for rawsocket
```

Comment:
yes, this is the websocket subprotocol (if the transport is over websocket): a string like `"wamp.2.json"` and similar